modules/afsocket/afunix-source.c

```diff
@@ -85,9 +86,20 @@ static gboolean
 afunix_sd_init(LogPipe *s)
 {
   AFUnixSourceDriver *self = (AFUnixSourceDriver *) s;
+  gboolean result;
+ 
+  result = afsocket_sd_init_method(s);
+  
+  if (is_file_socket(self->filename))
+    afunix_sd_apply_perms_to_socket(self);
+  else
+    {
+      msg_warning("Source is not a unix socket",
+                evt_tag_str("filename", self->filename),
+                NULL);
+    }
 
```

Comment:
Hmmm... this is not the best.
What if `afsocket_sd_init_method()` fails? We should return immediately.
And a code-style issue:
when using a block statement as one of the branches within an `if-else` statment, then every branch should be grouped into a block-statement, even if it contains only one statement.